src/test/java/fr/techad/edc/client/CommonBase.java

```diff
@@ -59,7 +60,8 @@ protected EdcReader createEdcReader() {
 
         DocumentationItemProvider documentationItemProvider = new DocumentationItemProvider();
         InformationProvider informationProvider = new InformationProvider();
-        this.edcReader = new HttpReaderImpl(httpClient, clientConfiguration, keyUtil, contextItemProvider, documentationItemProvider, informationProvider);
+        I18NProvider i18nProvider = new I18NProvider();
```

Comment:
Use Guice to configure this prodivder.
See EdcClientModule.java :   bind(DocumentationItem.class).toProvider(DocumentationItemProvider.class);

Remove tne new InformationProvider. This is a bad practise.

The reader has to be created by the injector engine and not by the hand (see line 49 in EdcClientModule.java)

In fact this method has to be removed

Reply:
I need these injections for the tests